server/configuration.go

```diff
@@ -89,6 +89,10 @@ func (p *Plugin) OnConfigurationChange() error {
 		return errors.Wrap(err, "failed to load plugin configuration")
 	}
 
+	if err := p.API.KVDelete(RootURLKey); err != nil {
```

Comment:
I am not sure exactly how this works, but isn't deleting (resetting/rediscovering) every time in OnConfigurationChange a bit expensive? It is my understanding that the OnConfigurationChange hook may be invoked on an unrelated configuration change.

Reply:
Indeed, it should be deleted only when the `domain` setting is changed. Do you know how to fetch the older configuration values in the `OnConfigurationChange` hook?

Reply:
During the `OnConfigurationChange` hook, you should have access to the current config stored in the plugin process right? You can do a diff between the two configs to find out if the `Domain` value changed.

Reply:
Thanks, done :+1: